### DISCUSSION TITLE
Project Progress

### DISCUSSION THREAD

**Discussion by dukelec** [dukelec/cd_pnp#2] [Announcements]
Recently completed:

The air pump has been mounted to the X-axis slide, with silicone damping, the vibration of the air pump can be ignored, and the air tube connection is much simpler.

The wires of X and Y axes are supported by springs, so they are no longer messy.

A fixed camera has been added to assist in nozzle calibration through an optical scale.

![DSC_0417_1](https://user-images.githubusercontent.com/4454365/168339943-a64fa933-0733-4db7-96f5-07a85d270356.JPG)

For the R-axis also added a detection switch for automatic zeroing after power on:

![Screenshot_2022-05-14_01-21-41](https://user-images.githubusercontent.com/4454365/168339992-38eb4734-f1cc-4d83-bf76-d3fef5146e94.png)

Added an acrylic top cover for the exposed circuit board.

Added a circle of light source for the fixed camera.

Added a small adapter board for CDBUS Bridge and CD-PnP connection to provide a socket for DC adapters.

Next:

Add some wire fixing clips on the base plate.

Design the package and start selling it in stock (probably in a few weeks).

**Comment by dukelec** [comment]
Updates:

The backlight panel moves to the bottom and the silicone pad can be flatter and larger.
(The silicone pad will be replaced with a 130x300x3.0 mm pad later.)

Calibration camera mounted on the base, more convenient to use.

![DSC_0593](https://user-images.githubusercontent.com/4454365/180499397-652b9b2a-5cef-4714-9f39-cbea0abba1b9.JPG)

**Reply by chenleigang** [reply]
This is a very good project, which is under constant attention

**Comment by dukelec** [comment]
Upgraded CDBUS-BRIDGE hardware to version 4.2 for CD-PnP, with STM32G0B1 inside:
(The follow picture is version 4.1)
![cd-bridge-top](https://user-images.githubusercontent.com/4454365/207029191-fae9c79f-1d34-4099-a64c-eb369491fe59.jpg)
![cd-bridge-bot](https://user-images.githubusercontent.com/4454365/207029201-a6672954-04e6-4e5b-9899-8d18a480aa39.jpg)

**Comment by dukelec** [comment]
Assembly in progress:
![DSC_0904](https://user-images.githubusercontent.com/4454365/211186710-bcbcbd69-c62b-47a8-82c0-d711e9098619.JPG)
